src/rbdmap

```diff
@@ -106,13 +97,80 @@ do_unmap() {
 	exit ${RET}
 }
 
+do_unmap_file() {
+	RET=0
+	## skip if nothing is mapped
+	ls /dev/rbd[0-9]* >/dev/null 2>&1 || exit ${RET}
+
+	# Read /etc/rbdtab to create non-existant mapping
+	while read DEV PARAMS; do
+		case "$DEV" in
+		  ""|\#*)
+			continue
+			;;
+		  */*)
+			;;
+		  *)
+			DEV=rbd/$DEV
+			;;
+		esac
+
+		MAP_RV="$(readlink -f /dev/rbd/$DEV)"
+		if [ ! -b $MAP_RV ]; then
+			logger -p "daemon.debug" -t rbdmap "$DEV not mapped, skipping unmap"
+			continue
+		fi
+
+		## pre-unmapping
+		if [ -x "/etc/ceph/rbd.d/${DEV}" ]; then
+			logger -p "daemon.debug" -t rbdmap "Running pre-unmap hook '/etc/ceph/rbd.d/${DEV}'"
+			/etc/ceph/rbd.d/${DEV} unmap "/dev/rbd/${DEV}"
+		fi
+
+		logger -p "daemon.debug" -t rbdmap "Unmapping '/dev/rbd/${DEV}'"
+		MNT=$(findmnt --mtab --source $MAP_RV --noheadings | awk '{print $1'})
+		if [ -n "${MNT}" ]; then
+		    logger -p "daemon.debug" -t rbdmap "Unmounting '${MNT}'"
+		    umount "${MNT}" >>/dev/null 2>&1
+		fi
+		if mountpoint -q "${MNT}"; then
+		    ## Un-mounting failed.
+		    logger -p "daemon.warning" -t rbdmap "Failed to unmount '${MNT}'"
+		    RET=$((${RET}+1))
+		    continue
+		fi
+		## Un-mapping.
+		rbd unmap /dev/rbd/${DEV} >>/dev/null 2>&1
+		if [ $? -ne 0 ]; then
+		    logger -p "daemon.warning" -t rbdmap "Failed to unmap '${DEV}'"
+		    RET=$((${RET}+$?))
+		    continue
+		fi
+		logger -p "daemon.debug" -t rbdmap "Unmapped '${DEV}'"
+	done < $RBDMAPFILE
+	exit ${RET}
+}
+
+# default to reasonable value if RBDMAPFILE not set in environment
+RBDMAPFILE="${RBDMAPFILE:-/etc/ceph/rbdmap}"
+
+if [ ! -f "$RBDMAPFILE" ]; then
+	logger -p "daemon.warning" -t rbdmap "No $RBDMAPFILE found."
+	exit 0
+fi
+
```

Comment:
oh, now I see that you didn't remove this bit of code, but rather generalized it so it triggers for both map and unmap